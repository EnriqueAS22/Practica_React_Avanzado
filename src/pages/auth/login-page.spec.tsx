import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import LoginPage from "./login-page";
import { Provider } from "react-redux";

describe("LoginPage", () => {
  const state = {
    auth: false,
    adverts: { data: [], loaded: false },
    ui: { pending: false, error: null },
  };
  const renderComponent = () =>
    render(
      <Provider
        store={{
          getState: () => state,
          // @ts-expect-error: subscribe
          subscribe: () => {},
          // @ts-expect-error: dispatch
          dispatch: () => {},
        }}
      >
        <LoginPage />
      </Provider>,
    );

  test("should render", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test("Should dispatch login action", async () => {
    renderComponent();
    const emailInput = screen.getByLabelText("Email");
    const passwordInput = screen.getByLabelText("Password");
    const button = screen.getByRole("button");
    expect(button).toHaveTextContent("Log in!");
    expect(button).toBeDisabled();

    await userEvent.type(emailInput, "user@mail.com");
    await userEvent.type(passwordInput, "1234");
    expect(button).toBeEnabled();
    await userEvent.click(button);
  });
});
