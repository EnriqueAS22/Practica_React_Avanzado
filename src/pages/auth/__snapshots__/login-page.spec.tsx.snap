// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > should render 1`] = `
<div>
  <div
    class="mt-6 flex justify-center"
  >
    <form
      class="space-y-12 rounded-2xl bg-white p-8 shadow-lg"
    >
      <div
        class="mt-10 grid grid-cols-1 gap-x-6 gap-y-8 sm:grid-cols-6"
      >
        <div
          class="sm:col-span-3"
        >
          <label
            class="block text-sm font-medium text-gray-700"
            for="email"
          >
            Email
          </label>
          <div
            class="mt-2"
          >
            <input
              class="block w-full rounded-lg border border-gray-300 bg-white px-4 py-2 text-base text-gray-900 shadow-sm focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
              id="email"
              name="email"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="mt-10 grid grid-cols-1 gap-x-6 gap-y-8 sm:grid-cols-6"
      >
        <div
          class="sm:col-span-3"
        >
          <label
            class="block text-sm font-medium text-gray-700"
            for="password"
          >
            Password
          </label>
          <div
            class="mt-2"
          >
            <input
              class="block w-full rounded-lg border border-gray-300 bg-white px-4 py-2 text-base text-gray-900 shadow-sm focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex items-center"
      >
        <input
          class="border-border text-primary focus:ring-primary h-4 w-4 rounded"
          id="remember"
          name="remember"
          type="checkbox"
        />
        <label
          class="text-text ml-2 block text-sm"
          for="remember"
        >
          Remember
        </label>
      </div>
      <button
        class="sc-bXdtCk hcOKsH login-form-submit"
        disabled=""
        type="submit"
      >
        Log in!
      </button>
    </form>
  </div>
</div>
`;
